playwright/support/setup.ts

```diff
@@ -0,0 +1,69 @@
+import {
+  IWorldOptions,
+  setWorldConstructor,
+  setDefaultTimeout,
+  World,
+  After,
+  Before,
+} from "@cucumber/cucumber";
+import {
+  Browser,
+  BrowserContext,
+  BrowserContextOptions,
+  chromium,
+  Page,
+} from "@playwright/test";
+
+interface PlaywrightParameters {
+  browser: Browser;
+  context: BrowserContext;
+  page: Page;
+}
+
+export class CustomWorld extends World {
```

Comment:
Can you explain a bit what's going on in this file (or share a link to docs)? A "World" object seems like a strange thing in the JS context.

Reply:
Sure thing. This is a link to the [cucumber-js docs](https://github.com/cucumber/cucumber-js/blob/main/docs/support_files/world.md) where it talks about the World scope. It's basically a Cucumber setup file. In our case, we need to extend the default cucumber-js world to include Playwright-specific browser objects.